tests/components/demo/test_vacuum.py

```diff
@@ -47,326 +47,293 @@
-
-        assert old_state_complete != new_state_complete
-        assert FAN_SPEEDS[1] == old_state_complete.attributes[ATTR_FAN_SPEED]
-        assert FAN_SPEEDS[0] == new_state_complete.attributes[ATTR_FAN_SPEED]
-
-        assert old_state_state != new_state_state
-        assert FAN_SPEEDS[1] == old_state_state.attributes[ATTR_FAN_SPEED]
-        assert FAN_SPEEDS[0] == new_state_state.attributes[ATTR_FAN_SPEED]
-
-    def test_send_command(self):
-        """Test vacuum service to send a command."""
-        group_vacuums = ",".join([ENTITY_VACUUM_BASIC, ENTITY_VACUUM_COMPLETE])
-        old_state_basic = self.hass.states.get(ENTITY_VACUUM_BASIC)
-        old_state_complete = self.hass.states.get(ENTITY_VACUUM_COMPLETE)
-
-        common.send_command(
-            self.hass, "test_command", params={"p1": 3}, entity_id=group_vacuums
-        )
-
-        self.hass.block_till_done()
-        new_state_basic = self.hass.states.get(ENTITY_VACUUM_BASIC)
-        new_state_complete = self.hass.states.get(ENTITY_VACUUM_COMPLETE)
-
-        assert old_state_basic == new_state_basic
-        assert old_state_complete != new_state_complete
-        assert STATE_ON == new_state_complete.state
-        assert (
-            "Executing test_command({'p1': 3})"
-            == new_state_complete.attributes[ATTR_STATUS]
-        )
+@pytest.fixture(autouse=True)
+async def setup_demo_vacuum(hass):
+    """Initialize setup demo vacuum."""
+    assert await async_setup_component(hass, DOMAIN, {DOMAIN: {CONF_PLATFORM: "demo"}})
+
+
+async def test_supported_features(hass):
+    """Test vacuum supported features."""
+    state = hass.states.get(ENTITY_VACUUM_COMPLETE)
+    assert state.attributes.get(ATTR_SUPPORTED_FEATURES) == 2047
+    assert state.attributes.get(ATTR_STATUS) == "Charging"
+    assert state.attributes.get(ATTR_BATTERY_LEVEL) == 100
+    assert state.attributes.get(ATTR_FAN_SPEED) == "medium"
+    assert state.attributes.get(ATTR_FAN_SPEED_LIST) == FAN_SPEEDS
+    assert state.state == STATE_OFF
+
+    state = hass.states.get(ENTITY_VACUUM_MOST)
+    assert state.attributes.get(ATTR_SUPPORTED_FEATURES) == 219
+    assert state.attributes.get(ATTR_STATUS) == "Charging"
+    assert state.attributes.get(ATTR_BATTERY_LEVEL) == 100
+    assert state.attributes.get(ATTR_FAN_SPEED) is None
+    assert state.attributes.get(ATTR_FAN_SPEED_LIST) is None
+    assert state.state == STATE_OFF
+
+    state = hass.states.get(ENTITY_VACUUM_BASIC)
+    assert state.attributes.get(ATTR_SUPPORTED_FEATURES) == 195
+    assert state.attributes.get(ATTR_STATUS) == "Charging"
+    assert state.attributes.get(ATTR_BATTERY_LEVEL) == 100
+    assert state.attributes.get(ATTR_FAN_SPEED) is None
+    assert state.attributes.get(ATTR_FAN_SPEED_LIST) is None
+    assert state.state == STATE_OFF
+
+    state = hass.states.get(ENTITY_VACUUM_MINIMAL)
+    assert state.attributes.get(ATTR_SUPPORTED_FEATURES) == 3
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5315232)